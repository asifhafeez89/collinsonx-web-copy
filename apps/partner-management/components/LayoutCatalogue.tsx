import colors from '@collinsonx/design-system/colour-constants-partner';
import { Box, Flex, MediaQuery } from '@collinsonx/design-system/core';
import Link from 'next/link';
import ContentWrapper from './ContentWrapper';

interface LayoutProps {
  subHeader?: JSX.Element;
  heading?: JSX.Element;
  disableWrapper?: boolean;
  children: JSX.Element;
}

export default function Layout({
  children,
  heading,
  disableWrapper = false,
  subHeader,
}: LayoutProps) {
  return (
    <div
      style={{
        width: '100%',
        height: '100%',
      }}
    >
      <MediaQuery
        query="print"
        styles={{
          display: 'none',
        }}
      >
        <header
          style={{
            backgroundColor: colors['bg-surface-dark'],
            borderBottom: `1px solid ${colors['grey-border']}`,
            width: '100%',
            padding: '2rem',
            textAlign: 'center',
            position: 'relative',
            margin: 0,
          }}
        >
          <Flex justify="center" align="center">
            <Box
              sx={{
                position: 'absolute',
                right: '60px',
              }}
            >
              <Link href="/auth/signout">Logout</Link>
            </Box>
          </Flex>
        </header>
      </MediaQuery>
      {subHeader && (
        <MediaQuery
          query="print"
          styles={{
            display: 'none',
          }}
        >
          {subHeader}
        </MediaQuery>
      )}

      {heading && (
        <div
          style={{
            padding: '0 64px',
            margin: 0,
            backgroundColor: colors.white,
            borderBottom: `1px solid ${colors['grey-border']}`,
          }}
        >
          {heading}
        </div>
      )}
      <main style={{ height: '100%' }}>
        {!disableWrapper ? (
          <ContentWrapper>{children}</ContentWrapper>
        ) : (
          children
        )}
      </main>
    </div>
  );
}
