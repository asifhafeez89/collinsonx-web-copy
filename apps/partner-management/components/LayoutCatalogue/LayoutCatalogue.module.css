.headerInner {
  position: absolute;
  right: 60px;
}

.content {
  max-height: 100%;
  max-width: 1140px;
  margin: 0 auto;
}
