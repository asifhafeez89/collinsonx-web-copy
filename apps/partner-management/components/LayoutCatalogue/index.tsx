import colors from '@collinsonx/design-system/colour-constants-partner';
import classes from './LayoutCatalogue.module.css';
import { Box, Flex } from '@collinsonx/design-system/core';
import { useMediaQuery } from '@collinsonx/design-system/hooks';
import Link from 'next/link';
import ContentWrapper from '../ContentWrapper';

interface LayoutProps {
  subHeader?: JSX.Element;
  heading?: JSX.Element;
  disableWrapper?: boolean;
  children: JSX.Element;
}

export default function Layout({
  children,
  heading,
  disableWrapper = false,
  subHeader,
}: LayoutProps) {
  const print = useMediaQuery('print');
  return (
    <div
      style={{
        width: '100%',
        height: '100%',
      }}
    >
      {print && (
        <>
          <header
            style={{
              backgroundColor: colors['bg-surface-dark'],
              borderBottom: `1px solid ${colors['grey-border']}`,
              width: '100%',
              padding: '2rem',
              textAlign: 'center',
              position: 'relative',
              margin: 0,
            }}
          >
            <Flex justify="center" align="center">
              <Box className={classes.headerInner}>
                <Link href="/auth/signout">Logout</Link>
              </Box>
            </Flex>
          </header>
          {subHeader}
        </>
      )}

      {heading && (
        <div
          style={{
            padding: '0 64px',
            margin: 0,
            backgroundColor: colors.white,
            borderBottom: `1px solid ${colors['grey-border']}`,
          }}
        >
          {heading}
        </div>
      )}
      <main style={{ height: '100%' }}>
        {!disableWrapper ? (
          <ContentWrapper>{children}</ContentWrapper>
        ) : (
          children
        )}
      </main>
    </div>
  );
}
