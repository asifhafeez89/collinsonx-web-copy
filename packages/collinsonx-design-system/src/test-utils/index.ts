import userEvent from '@testing-library/user-event';

export * from '@testing-library/react';
export { userEvent };
export { default as render, Provider } from './render';
