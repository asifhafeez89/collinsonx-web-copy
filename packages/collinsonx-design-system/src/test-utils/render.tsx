import { RenderOptions, render } from '@testing-library/react';
import { MantineProvider } from '@mantine/core';
import { ReactElement, ReactNode } from 'react';

export const Provider = ({ children }: { children: ReactNode }) => (
  <MantineProvider>{children}</MantineProvider>
);

type Render = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => ReturnType<typeof render>;

const customRender: Render = (ui, options) =>
  render(ui, {
    wrapper: Provider as unknown as RenderOptions['wrapper'],
    ...options,
  });

export default customRender;
