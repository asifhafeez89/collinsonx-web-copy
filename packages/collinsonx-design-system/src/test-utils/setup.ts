import userEvent from '@testing-library/user-event';
import render from './render';

export default function (jsx: Parameters<typeof render>[0]) {
  return {
    user: userEvent.setup(),
    ...render(jsx),
  };
}
