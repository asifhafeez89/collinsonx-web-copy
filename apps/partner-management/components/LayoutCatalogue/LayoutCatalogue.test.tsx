import LayoutCatalogue from '.';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@collinsonx/design-system/test-utils';

describe('<LayoutCatalogue />', () => {
  it('should render', () => {
    const component = render(
      <LayoutCatalogue subHeader={<>header</>} heading={<>heading</>}>
        <></>
      </LayoutCatalogue>
    );

    expect(component).toMatchSnapshot();
  });
});
