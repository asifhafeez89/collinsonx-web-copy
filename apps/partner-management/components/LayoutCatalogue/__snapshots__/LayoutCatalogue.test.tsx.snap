// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LayoutCatalogue /> should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <style
        data-mantine-styles="classes"
      >
        @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
      </style>
      <div
        style="width: 100%; height: 100%;"
      >
        <div
          style="padding: 0px 64px; margin: 0px; background-color: rgb(255, 255, 255); border-bottom: 1px solid #D5D5D5;"
        >
          heading
        </div>
        <main
          style="height: 100%;"
        >
          <div
            class="contentWrapper"
          />
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <style
      data-mantine-styles="classes"
    >
      @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
    </style>
    <div
      style="width: 100%; height: 100%;"
    >
      <div
        style="padding: 0px 64px; margin: 0px; background-color: rgb(255, 255, 255); border-bottom: 1px solid #D5D5D5;"
      >
        heading
      </div>
      <main
        style="height: 100%;"
      >
        <div
          class="contentWrapper"
        />
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
